Sendmail rules: Extend filter by check-in state (#3489)
pretix/pretix: doc/api/resources/sendmail_rules.rst

                                                                 ``n__valid_if_pending`` for payment pending but already confirmed,
                                                                 and ``n__pending_overdue`` for pending with payment overdue.
                                                                 The default is ``["p", "n__valid_if_pending"]``.
checked_in_status                     string                     Checked in status to restrict recipients to. Valid strings

>                                                                  are: ``all`` for no filtering (default), ``checked_in`` for limiting to

Reviewer: I thought about that too, although I thought of `"ignore"` instead of `null`. I do like `null` better though – this is more similar to just leaving the query-param out completely (which works as well).

>                                                                  are: ``all`` for no filtering (default), ``checked_in`` for limiting to

Author: This was a bigger change than anticipated, but I agree with you both and changed it to accept `null`

> checked_in_status                     string                     Checked in states that can be used to filter recipients. Valid strings
>                                                                  are: ``all`` (no filtering, default), ``checked_in`` for
>                                                                  sending mails to checked in customers, and ``no_checkin`` to send mails
>                                                                  to all customers who have not checked in.

Reviewer: Not really an issue, I would make the wording and formatting (use of paretheses) more consistent.

```suggestion
checked_in_status                     string                     Checked in status to restrict recipients to. Valid strings
                                                                 are: ``all`` for no filtering (default), ``checked_in`` for limiting to
                                                                 attendees that are or have been checked-in, and ``no_checkin`` for
                                                                 limiting to attendees who have not checked-in.
```

> checked_in_status                     string                     Checked in status to restrict recipients to. Valid strings
>                                                                  are: ``all`` for no filtering (default), ``checked_in`` for limiting to
>                                                                  attendees that are or have been checked-in, and ``no_checkin`` for
>                                                                  limiting to attendees who have not checked-in.

Reviewer 2: ```suggestion
checked_in_status                     string                     Check-in status to restrict recipients to. Valid strings
                                                                 are: ``all`` for no filtering (default), ``checked_in`` for limiting to
                                                                 attendees that are or have been checked in, and ``no_checkin`` for
                                                                 limiting to attendees who have not checked in.
```
